### PR TITLE
Use fast-vector highlighting and force-merge after indexing

### PR DESCRIPTION
**Pull request by yrodiere**
I couldn't reproduce the same amount of slowness local as in prod, but I suspect hardware might be to blame... Heap size for OpenSearch -- as well as the app --  is the same as in prod.

In any case, search was still relatively slow on my machine at ~100ms per query. Disabling highlighting brought me to 10ms, proving that, as expected, highlighting is to blame.

So I switched to the fast-vector highlighter and saw a small improvement: instead of ~100ms, queries now take ~50 to ~70ms (still on my laptop). I had to remove stop-words filters and change the boundary scanner, but otherwise highlighted snippets seem similar.

Hopefully the improvement will be similar in prod.

I also added a force-merge after indexing because... why not. I experimented with concurrent fragment search, but this didn't provide a significant speedup.

### CODE REVIEW

File: src/test/java/io/quarkus/search/app/SearchServiceTest.java [hunk 1]
```diff
@@ -217,8 +217,8 @@ private static List<Arguments> relevance() {
                         GuideRef.HIBERNATE_REACTIVE,
                         GuideRef.HIBERNATE_REACTIVE_PANACHE,
                         GuideRef.HIBERNATE_ORM_PANACHE,
-                        GuideRef.HIBERNATE_ORM_PANACHE_KOTLIN,
-                        GuideRef.HIBERNATE_ORM)),
+                        GuideRef.HIBERNATE_ORM,
+                        GuideRef.HIBERNATE_ORM_PANACHE_KOTLIN)),
```
Comment (marko-bekhta) [inline src/test/java/io/quarkus/search/app/SearchServiceTest.java:221]: 😏 sooooooooo does ORM guide contain a lot of stop words? 😃 🙈

**Review by marko-bekhta — APPROVED** [review]
🏆 🎉

File: src/main/java/io/quarkus/search/app/SearchService.java [hunk 2]
```diff
@@ -108,21 +108,24 @@ public SearchResult<GuideSearchHit> search(@RestQuery @DefaultValue(QuarkusVersi
...
+                            // We want the whole text of the field, regardless of whether it has a match or not.
+                            .noMatchSize(TITLE_OR_SUMMARY_MAX_SIZE)
+                            .fragmentSize(TITLE_OR_SUMMARY_MAX_SIZE)
+                            // We want the whole text as a single fragment
+                            .numberOfFragments(1))
+                    .highlighter("highlighter_content", f -> f.unified()
+                            // If there's no match in the full content we don't want to return anything.
+                            .noMatchSize(0)
+                            // Content is really huge, so we want to only get small parts of the sentences.
+                            .orderByScore(true)
```
[the remark this answers is not in the document]
Comment (yrodiere) [reply src/main/java/io/quarkus/search/app/SearchService.java]: I gave it a try, and it does speed things up by another 10-20 ms on my laptop... BUT it results in much worse highlighting in some scenarios.
For example if you search for `quarkus.websocket.max-frame-size`, we'll have matches on `quarkus.websocket.max-frame-size` (all terms matching), but also on "initial-**size**" (only one term matching). Without scopring, the highlights end up containing only this second, irrelevant result :/

[the remark this answers is not in the document]
Comment (yrodiere) [reply src/main/java/io/quarkus/search/app/SearchService.java]: So, I didn't do it :(

**Comment by yrodiere** [comment]
> Ok, I forgot to run the synonyms tests, I'll update those next week.

Should be good now.

File: src/main/java/io/quarkus/search/app/SearchService.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (marko-bekhta) [reply src/main/java/io/quarkus/search/app/SearchService.java]: Thanks for giving it a try 🏆

**Pull request merged**